Feat/animate user answer

casaper opened the pull request:
Preparing review for this afternoon

casaper committed:
chore: rename class and delete rubble

casaper committed:
chore: loatds of different cleanup things

- rename of components
- rename of components filenames according to components name
- no more destruction props for better readability
- removed className props
- ...

casaper committed:
chore: removed rubble files

casaper committed:
style(classNames): adapted classNames to be more convenient

casaper committed:
chore: simplified bot part

casaper committed:
feat: simplified bot part past rendering

casaper committed:
style: minor

casaper committed:
chore: removed rubble tests

casaper committed:
chore: lo-dash added to package

casaper committed:
feat: bot text now singular and not an array anymore

casaper committed:
chore: loads of simplifications and cleanup

casaper committed:
feat: handle random bot part with lodash in callback

casaper committed:
chore: minor class naming changes

casaper committed:
style(): aligning and such minor ccode style things

casaper committed:
chore: removed unknown prop

some leftover from some past experiment

casaper committed:
feat: starting to replace sass with inline styles

casaper committed:
chore: rename components to User*Past

casaper committed:
chore: next inline style step

casaper committed:
chore: user buton generalized to one component

- past and disabled and present are now handled with one component

casaper committed:
feat: more inline styles

casaper committed:
feat: more inline styles

casaper committed:
feat: finalizing inline styles, remove scss

casaper committed:
style: minor indent stuff

casaper committed:
chore: removed last unneeded classNames

casaper committed:
chore: minor sample test ignored

casaper committed:
feat: from inject to var paradigm

casaper committed:
style(code): minor code style mangling

casaper committed:
fix: scrolling to actual bubbles working again

casaper committed:
fix: updated the new vals names

casaper committed:
ignore npm debug log

casaper committed:
feat(styles): some copycat from uxdings

casaper committed:
feat(styles): bubble margins

casaper committed:
feat: bot part converted to statefull component

casaper committed:
feat(styles): disabled button different

casaper committed:
started hels conversation

casaper committed:
chore: removed unneeded files

casaper committed:
feat: loads of thing in order to introduce simplified Bubble component

- <Bubble> component
- more simplified styles
- some namings changed
- better history log with no JSON.stringify use
- misc other things

casaper committed:
chore: changed the path to stepId naming

casaper committed:
chore: better callbback naming

casaper committed:
style(codestyle): mainly indents and such

casaper committed:
feat(style): some user bubble color and margins/paddings

casaper committed:
fix(styles): some style fixing - center box and such

casaper committed:
minor style

casaper committed:
feat: reduced 3 callbacks to one

casaper committed:
feat: functional styles for main

casaper committed:
style(): minor indents and such

casaper committed:
chore: some renaming and always define template varData

casaper committed:
chore: "in react unitless ints are px" - ...

casaper committed:
chore: removed obsolete component

casaper committed:
chore: why not pass the button number, instead of defaulting it in Main

casaper committed:
fix: give the bot a name instead of id

casaper committed:
fix: past user input shown again in past bubbles

casaper committed:
chore: removed obsolete Style obj

casaper committed:
feat: trying to get bubble dripping, but I'm mislead

or I'd bether say, this isn't the right way, but I'm commiting it
for historic purposes

casaper committed:
expeeriment: a little better, but still a mess

casaper committed:
experiment: now it seems to kinda work...

casaper committed:
style: indents and namings

casaper committed:
fix(styles): implemented review recommendations

- there is now only one parameter needed for past user bubbles
the past part thingie is now gone

casaper committed:
feat: the icon component is obsolete now

- removed icon component because it is only used inside bot part

casaper committed:
style: opptimized styles and moved to more readable loccations

casaper committed:
fix: if bubble is type forward it is invisible

casaper committed:
feat: active bubbles are called with type speaking

in order to cause animations to be launched inside the bubble component

casaper committed:
feat: some animations in place and seem to be working

casaper committed:
experiment: animation, but not as it should be

casaper committed:
experiment: more mangling arround with animations and transitions....

casaper committed:
style: minor wrong comma

casaper committed:
chore: del debug log

casaper committed:
fix: dev stack hot reloading

- the webpack dev stack seemed not to be working at the momment
- as far as I found out it is a general but that appeared just
  lateley - I assume it will be fixed some time soon
- so for now there is no hot reloading in the dev server

casaper committed:
chore: disable unneeded loaders

casaper committed:
chore: cleanup package.json

- storybook newly added
- babel-preset-es2017 now in devDependencies, as it should be
- phantomjs-prebuilt is not needed
- lodash is not needed anymore

casaper committed:
feat: introducing yarn with its lock file and with peerDependencies

casaper committed:
fix: removed missing babel-plugin

casaper committed:
chore: moved storybooks stories directory to src

casaper committed:
style: explanatory comment about disabling hot reloading

casaper committed:
chore(minor): storybook mooved and misc (minor stuff)

casaper committed:
chore(minor): ignoreing backup package.json

casaper committed:
style: some comments and indents

casaper committed:
feat(styles): made styles functional

casaper committed:
style: ...

koffeinfrei commented on cfg/defaults.js:
```diff
@@ -32,22 +32,22 @@ function getDefaultModules() {
+      //{
+      //  test: /\.styl/,
+      //  loader: 'style-loader!css-loader!stylus-loader'
+      //},
```
remove uncommented code

koffeinfrei commented on dist/index.html:
```diff
@@ -6,6 +6,78 @@
   <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
   <meta name="description" content="">
   <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
+  <style>
```
Extract to css files. You can use sass as well.

koffeinfrei commented on src/components/Bubble.js:
```diff
@@ -0,0 +1,99 @@
+  if(type === 1) {
+    return children;
+  }
+}
```
remove, obsolete

koffeinfrei commented on src/components/Bubble.js:
```diff
@@ -0,0 +1,99 @@
+    opacity: 1
+  }
+}
+const style = (type) => {
```
ask marco if those functions should be scope to `Bubble`.

koffeinfrei commented on src/components/Main.js:
```diff
@@ -1,211 +1,148 @@
-      });
-      this.setState({path, templateVars});
+  nextStepCb({answerBtnNo, userTxtInput}) {
+    answerBtnNo = (typeof answerBtnNo === 'undefined') ? 0 : answerBtnNo; // most likeley to be removed, just as a quick ensurance for it to be 0
```
`answerBtnNo || 0`

koffeinfrei commented on src/components/Main.js:
```diff
@@ -1,211 +1,148 @@
+                  key={bubbleIndex}
+                  type={(bubbleIndex === stepBotTexts.length-1) ? 'speaking': 'default'}
+                  answerPresent={answer}>
+                  {eval('`' + botText + '`')}
```
use some sort of string replace function / templating.

koffeinfrei commented on src/components/Main.js:
```diff
@@ -1,211 +1,148 @@
-            },
-            style: this.answerStyle
-          }} />
+        <div style={style('conversationPart')} id='scrollTarget'>
```
consider using `ref={...}` instead of id attribute.

koffeinfrei commented on src/components/PastPart.js:
```diff
@@ -0,0 +1,50 @@
+import BotPart          from './BotPart.js';
+import Bubble from './Bubble.js';
+
+const Styl = {
```
-> `Style`

koffeinfrei commented on src/components/PastPart.js:
```diff
@@ -0,0 +1,50 @@
+    case 'input':
+      userBubbleTxt = userTxtInput[step.inputProperty];
+      break;
+  }
```
extract to function `getUserBubbleText` or similar.

koffeinfrei commented on src/components/PastPart.js:
```diff
@@ -0,0 +1,50 @@
+          );
+        })}
+      </BotPart>
+      <Bubble key={'user_'+stepIndex} type={(type !== 'forward') ? 'past' : 'invisible'}>
```
Use `===`

koffeinfrei commented on src/components/UserAnswerPart.js:
```diff
@@ -0,0 +1,63 @@
+              return <Bubble key={answerBtnNo}  type={type}> { text } </Bubble>;
+
+            case 'forward': // Timout either from json or 2s -> leads to next step
+              const timeoutTime = (time) ? time : 2000;
```
`time || 2000`

koffeinfrei commented on dist/index.html:
```diff
@@ -6,6 +6,78 @@
+  </style>
 </head>
 <body>
   <div id="app">APPLICATION CONTENT</div>
```
git ignore `/dist`

casaper committed:
chore(review): remove commented config option

- there was a problem with react hot loader

casaper committed:
chore: move styles in seperate scss file again

casaper committed:
chore: rename answer boolean to speaking showAnswer

casaper committed:
chore: split style function up in seperate smaller functions

casaper committed:
chore: ignore dist directory at least for the future

casaper committed:
chore: nicer userBubbleText getter

- but not as suggested in review, because lifting the stuff out of the inner scope would
be a mess as well, pretty much

casaper committed:
style: some comment and coding style stuff

casaper committed:
chore: remove some rubble

casaper committed:
chore: doing the style named propperly and functional

casaper committed:
style: nicer object mergeing

casaper committed:
chore: rename to better nextStepCallback

casaper committed:
chore: better style funceiton

casaper committed:
fix: correct forgotten renaming and simplify code

casaper committed:
style: little comment

casaper committed:
chore: removed obsolete comment
